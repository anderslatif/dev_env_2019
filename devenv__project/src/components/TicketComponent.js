/*eslint-disable*/
import React, { Component } from "react";

class TicketComponent extends Component {
    render(props) {
        let {
            ticket, 
            qrAction
        } = this.props;
        return(
            <div className="ticket__component">
                <div className="ticket__component--wrapper">
                    <div className="ticket__referencecode">
                        <h2>Referene code: {ticket.reference}</h2>
                    </div>
                    <div className="ticket__departureLocation">
                        <div className="ticket__departureLocation--identifier">A</div>
                        <p>Departure: {ticket.order_departure_location}</p>
                    </div>
                    <div className="ticket__arivalLocation">
                        <div className="ticket__arivalLocation--identifier">B</div>
                        <p>Arival: {ticket.order_arival_location}</p>
                    </div>
                    <div className="ticket__status">
                        <p>Status: {ticket.ticket_status}</p>
                    </div>
                    <div className="generate__qrdivision">
                        <button 
                            className="ticket__generateqr" 
                            name={ticket.reference}
                            onClick={qrAction}
                        >
                        Generate QR
                        </button>                    
                    </div>
                </div>
            </div>
        );
    }
}

export default TicketComponent;