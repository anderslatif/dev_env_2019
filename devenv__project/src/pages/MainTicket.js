/*eslint-disable*/
import React, { Component } from "react";
import TicketComponent from "../components/TicketComponent";
import GeneratedQRBlockComponent from "../components/GeneratedQRBlockComponent";
import {
    BrowserView,
    MobileView,
    isBrowser,
    isMobile
  } from "react-device-detect";
import axios from "axios";


const testOrdersLists = [
    {
        id:1,
        reference: "REF123456789",
        name: "name1",
        description: "description1",
        order_quantity: 45,
        order_departure_location: "Copenhagen, Denmark",
        order_arival_location: "Lisbon, Portugal",
        order_departure_date: "9.9.2045",
        order_arival_date: "9.9.2045",
        order_chemical_types: [
            {
                id: "id1",
                chemical: "H20"
            },
            {
                id: "id2",
                chemical: "HSO4"
            }
        ],
        ticket_status: "pending"
    }
];

class MainTicket extends Component {
    constructor(props) {
        super(props)
        this.state = {
            activeReference: "",
            qrblock: false,
            order: [],
            orderId: ""
        }
    }
    
    // set the active reference codes
    getReferenceCodes = (ev) => {
        const referenceCodes = ev.target.name;
        this.setState({activeReference: referenceCodes});
        // const referenceCodes = ev.target.getAttribute("data-reference");
    }

    // toggle qr block state component
    toggleQR = () => {
        this.setState((prevState) => ({qrblock: !prevState.qrblock}))
    }

    // full qr block action function
    qrAction = (ev) => {
        this.getReferenceCodes(ev);
        this.toggleQR(ev)
    }

    // close qr code block component
    closeQR = () => {
        this.setState({qrblock: false})
    }

    componentDidMount() {
        axios.get("https://localhost:8000/orders")
             .then(response => {
                    console.log("main_ticket_response: ", response.data[0].id)
                    this.setState(() => {
                        return {
                            orderId: response.data[0].id,
                            order: response.data
                        }
                    })
             })
             .catch(error => console.log("main_ticket_error: ", error))
    }
    render() {
        return(
            <MobileView>
            <div className="main__ticket">
                <button className="menus">Menus</button>
                <div className="main__ticket--wrapper">
                    <div className="orders__header">
                        <h2>Ticket</h2>
                        <input type="text" placeholder="Search order" name="inbound_gate_officer_search_order" />
                    </div>

                    <div className="orders__wrapperMobile">
                        {
                            this.state.order.map((ticket) => {
                                return <TicketComponent
                                            key={ticket.id}
                                            ticket={ticket}
                                            qrAction={this.qrAction}
                                            orderId={this.state.orderId}
                                       />
                            })
                        }
                    </div>
                </div>
                {
                    this.state.qrblock
                    &&
                    <GeneratedQRBlockComponent
                        referenceCodes={this.state.activeReference}
                        closeQR={this.closeQR}
                    />
                }
            </div>
            </MobileView>
        );
    }
}

export default MainTicket;