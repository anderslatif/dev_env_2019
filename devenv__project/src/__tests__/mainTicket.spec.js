/*eslint-disable*/
import React from "react";
import MainTicket from "../pages/MainTicket";
import Enzyme, { shallow, mount } from "enzyme";
import TicketComponent from "../components/TicketComponent";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({
    adapter: new Adapter()
});

describe("Test ticket function", () => {
    it("should change the qr state", () => {
        const mainTicket = shallow(<MainTicket />);

        let state = false;
        state = true;
        
        expect(state).toBe(true);
    });

    it("Should check if the order exists", () => {
        const wrapper = shallow(<MainTicket />);
        expect(wrapper.find(".orders__wrapperMobile").length).toBe(1);
    });

    it("Should create a snapshot", () => {
        const wrapper = shallow(<MainTicket />);
        expect(wrapper).toMatchSnapshot();
    });
});