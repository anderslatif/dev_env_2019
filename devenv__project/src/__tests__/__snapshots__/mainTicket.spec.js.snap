// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ticket function Should create a snapshot 1`] = `ShallowWrapper {}`;
